src/game/WorldHandlers/Spell.cpp

```diff
@@ -3388,6 +3388,9 @@ void Spell::finish(bool ok)
     // Stop Attack for some spells
     if (m_spellInfo->HasAttribute(SPELL_ATTR_STOP_ATTACK_TARGET))
         { m_caster->AttackStop(); }
+
+    if(!m_caster->GetMapId())
+        return;
```

Comment:
This does nothing, remove.
Or if it is in zero and there is code below it then this needs another look.